GeoLib/AnalyticalGeometry.cpp

```diff
@@ -207,26 +209,19 @@ bool lineSegmentIntersect(
 }
 
 bool lineSegmentsIntersect(const GeoLib::Polyline* ply,
-                           std::size_t &idx0,
-                           std::size_t &idx1,
+                           GeoLib::Polyline::SegmentIterator &seg_it0,
+                           GeoLib::Polyline::SegmentIterator &seg_it1,
                            GeoLib::Point& intersection_pnt)
 {
```

Comment:
The segment iterators `seg_it0` and `seg_it1` are employed for returning the information which segments are intersecting. They are set within the function and thus it is ensured that they belong to `ply`.